react-components/src/components/RevealToolbar/RuleBasedOutputsButton.tsx

```diff
@@ -22,35 +22,57 @@ import { useTranslation } from '../i18n/I18n';
 import { useFetchRuleInstances } from '../RuleBasedOutputs/hooks/useFetchRuleInstances';
 import { type AssetStylingGroup } from '../..';
 
+type EmptyRule = {
+  rule: any;
+  isEnabled: boolean;
+};
 type RuleBasedOutputsButtonProps = {
   onRuleSetStylingChanged?: (stylings: AssetStylingGroup[] | undefined) => void;
+  onRuleSetSelectedChanged?: (ruleSet: RuleAndEnabled | EmptyRule | undefined) => void;
 };
 export const RuleBasedOutputsButton = ({
-  onRuleSetStylingChanged
+  onRuleSetStylingChanged,
+  onRuleSetSelectedChanged
 }: RuleBasedOutputsButtonProps): ReactElement => {
   const [currentRuleSetEnabled, setCurrentRuleSetEnabled] = useState<RuleAndEnabled>();
-  const [ruleInstances, setRuleInstances] = useState<RuleAndEnabled[]>();
+  const [emptyRuleSelected, setEmptyRuleSelected] = useState<EmptyRule>();
+  const [ruleInstances, setRuleInstances] =
+    useState<Array<RuleAndEnabled | EmptyRule | undefined>>();
```

Comment:
Does it make sense that this is an array of undefined's? Could it be something like `Array<RuleAndEnabled | EmptyRule> | undefined` instead?

Reply:
https://github.com/cognitedata/reveal/pull/4453/commits/a1d04afb98d76e7e109372cd20be758f5ec689c6